src/OneSignal.ts

```diff
@@ -471,16 +487,16 @@ export default class OneSignal {
     if (emailProfile.emailId) {
       await OneSignalApi.updatePlayer(appId, emailProfile.emailId, {
         tags: tags,
-        email_auth_hash: emailProfile.emailAuthHash,
+        identifier_auth_hash: emailProfile.identifierAuthHash,
       });
     }
 
-    var { deviceId } = await Database.getSubscription();
+    const { deviceId } = await Database.getSubscription();
     if (!deviceId) {
       await awaitSdkEvent(OneSignal.EVENTS.REGISTERED);
     }
     // After the user subscribers, he will have a device ID, so get it again
-    var { deviceId: newDeviceId } = await Database.getSubscription();
+    const { deviceId: newDeviceId } = await Database.getSubscription();
     await OneSignalApi.updatePlayer(appId, newDeviceId!, {
       tags: tags
```

Comment:
I imagine we should also add `external_id_auth_hash` here correct @jkasten2 ?

Reply:
yes, confirmed you have added it now